e2e/engine/cmd.go

```diff
@@ -286,7 +254,60 @@ func (e *engineCmd) Run(ctx context.Context, eventC chan *node.Event) error {
 			}
 		}
 	}
+	if e.conf.CheckAppHashOnExit {
```

Comment:
I think we should run checkapphash() every test command not just on exist, so that we can better detect the command that breaks consensus.